src/pilot_outfit.c

```diff
@@ -828,6 +828,10 @@ void pilot_fillAmmo( Pilot* pilot )
    Outfit *o, *ammo;
 
    for (i=0; i<pilot->noutfits; i++) {
+      /* Must have outfit */
+      if (pilot->outfits[i] == NULL)
```

Comment:
Is there any case where this is expected to be `NULL`? Or is any instance in which this condition is satisfied evidence of a bug elsewhere?

Reply:
I haven't a clue. All I know is:

1. The game segfaulted at the next statement.
2. I don't think pilot was NULL.

This was just a shot in the dark. I can't check whether or not it worked because the segfault was very rare. :/ If I get the segfault again I'll remove the check and see if I can figure out the check or fix actually needed. (I figure the worst case scenario, an unnecessary check, isn't too bad.)

Reply:
Additionally, I think this will just move the segfault elsewhere. Grepping for `->noutfits`, you can find other similar loops without this test.

Reply:
I can revert it, that's fine.